src/Microsoft.Extensions.Configuration.Ini/IniConfigurationExtensions.cs

```diff
@@ -75,8 +75,12 @@ public static IConfigurationBuilder AddIniFile(this IConfigurationBuilder builde
 
             if (provider == null && Path.IsPathRooted(path))
             {
-                provider = new PhysicalFileProvider(Path.GetDirectoryName(path));
-                path = Path.GetFileName(path);
+                string pathToFile;
+                provider = FileConfigurationProvider.ResolvePhysicalFileProvider(path, out pathToFile);
```

Comment:
Yeah good idea, that duplication has always bugged me, moved it into an instance method on FileConfigurationSource that we call instead.